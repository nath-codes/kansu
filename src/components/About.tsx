import React from 'react'
import { Section } from './Section'
import { Text } from './Text'
import { Heading } from './Heading'
import { Container } from './Container'

export const About = () => (
  <Section>
    <Heading>About</Heading>
    <Container>
      <Text>
        <p>
          Kansu is myself, Nathan Tredgett. A freelance, senior frontend
          JavaScript developer based in Bristol, England.
        </p>
        <p>
          With 10 years experience in senior roles, I specialise in React and
          React-based frameworks, although I have commercial experience in most
          JS frameworks.
        </p>
        <p>
          My knowledge and experience allows me to consistently deliver
          effective, well-tested code, quickly. I work with teams of any size to
          create applications that are fast, consistent and accessible across
          any and all devices.
        </p>
      </Text>
    </Container>
  </Section>
)
