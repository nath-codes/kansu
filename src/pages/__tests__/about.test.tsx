import 'jest-styled-components'
import React from 'react'
import { renderWithTheme } from '../../components/__tests__/helpers'
import About from '../about'

jest.mock('../../components/seo')

describe('About', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<About />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
