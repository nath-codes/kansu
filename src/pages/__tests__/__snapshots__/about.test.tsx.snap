// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
Array [
  <div>
    About
  </div>,
  .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 3rem;
}

.c1 {
  color: #ffffff;
  font-family: "Roboto",sans-serif;
  font-size: 2.25rem;
  left: 0;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  line-height: 1;
  margin: 0;
  padding-right: 0.75rem;
  position: absolute;
  text-transform: uppercase;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c1:after {
  color: #ffffff;
  content: 'â– ';
  bottom: 0.25rem;
  font-size: 0.75rem;
  position: absolute;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  position: relative;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 1.125rem;
  line-height: 1.25;
}

.c3 a {
  color: #ffffff;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c3 a:hover:after {
  border-bottom: solid 0.125rem #ffffff;
  content: '';
  position: absolute;
  bottom: -0.125rem;
  left: 0;
  right: 0;
}

.c3 p {
  margin-bottom: 1.5rem;
}

.c3 p:last-child {
  margin-bottom: 0;
}

@media only screen and (min-width:768px) {
  .c2 {
    max-width: 37.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c2 {
    max-width: 40.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c1:after {
    bottom: 0;
    font-size: 1.5rem;
  }
}

@media only screen and (min-width:768px) {
  .c3 {
    line-height: 1.3333333333;
  }
}

@media only screen and (min-width:980px) {
  .c3 {
    margin-top: 1.5rem;
  }
}

<section
    className="c0"
  >
    <h2
      className="c1"
      style={
        Object {
          "marginTop": "-1.5rem",
          "opacity": 0,
          "top": "0",
        }
      }
    >
      About
    </h2>
    <div
      className="c2"
    >
      <div
        className="c3"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(1.5rem)",
          }
        }
      >
        <p>
          Kansu is myself, Nathan Tredgett. A full-stack JavaScript developer from Bristol, England.
        </p>
        <p>
          I've been a developer for over a decade and have worked on all types of projects: big to small, simple to complex, inspiring to the functional.
        </p>
        <p>
          I create code that is clean, elegant and concise, I love what I do and I am available to hire for projects across the UK.
        </p>
      </div>
    </div>
  </section>,
]
`;
